cli/azd/pkg/project/service_target_containerapp.go

```diff
@@ -80,8 +83,14 @@ func (at *containerAppTarget) Deploy(
 		return nil, fmt.Errorf("validating target resource: %w", err)
 	}
 
-	// Login, tag & push container image to ACR
-	_, err := at.containerHelper.Deploy(ctx, serviceConfig, packageOutput, targetResource, true, progress)
+	var err error
+
+	if serviceConfig.Docker.RemoteBuild {
+		err = at.remoteBuildHelper.RunRemoteBuild(ctx, serviceConfig, targetResource, progress)
```

Comment:
Maybe I should just merge `RunRemoteBuild` and this `if` into deploy.  AKS would get it for free then.  I think this worried me in the past, but does seem to make more sense now.